app/scripts/lib/transaction/swaps.ts

```diff
@@ -0,0 +1,122 @@
+import { pickBy, merge } from 'lodash';
+import {
+  TransactionStatus,
+  TransactionType,
+  TransactionMeta,
+} from '../../../../shared/constants/transaction';
+import MetamaskController from '../../metamask-controller';
+
+type SwapOptions = {
+  // Whether this transaction required an approval transaction
+  hasApproveTx: boolean;
+  // Additional metadata to store for the transaction
+  meta: any;
+};
+
+type SwapApprovalTransaction = {
```

Comment:
Is this type required? The API method on the `TransactionController` doesn't use a type so do we gain anything by casting internally?

Reply:
This file is removed